<?xml version="1.0" ?>
<launch>

  <!-- send table.xml to param server -->
  <arg name="map" default="/home/zwu/LIDAR-DATA/3oct-ds.pcd" />
  <arg name="namespace" default="bedok_localization" /> <!-- set to do multiple mapping at once -->
  <arg name="output_directory" default="$(env HOME)/ndt_custom/$(arg namespace)/" />

  <!-- ndt algorithm params -->
  <arg name="resolution" default="2.5"/> 
  <arg name="step_size" default="0.05"/>
  <arg name="transformation_epsilon" default="0.0001"/>
  <arg name="max_iteration" default="100" />
  <arg name="voxel_leaf_size" default="0.1" />
  <arg name="min_scan_range" default="3.2" />
  <arg name="min_add_scan_shift" default="0.5" />  
  <arg name="min_add_scan_yaw_diff" default="0.013" />

  <!-- initial pose TODO: add GNSS when available T.T -->
  <arg name="initial_x" default="-70.2561" />
  <arg name="initial_y" default="10.4625" />
  <arg name="initial_z" default="2.9815" />
  <arg name="initial_roll" default="-0.0312" />
  <arg name="initial_pitch" default="-0.0421" />
  <arg name="initial_yaw" default="2.8115" />
  
  <!-- tf from lidar frame to car frame -->
  <arg name="tf_x" default="0.0"/>
  <arg name="tf_y" default="0.0"/>
  <arg name="tf_z" default="0.0"/>
  <arg name="tf_roll" default="0.0"/>
  <arg name="tf_pitch" default="0.0"/>
  <arg name="tf_yaw" default="0.0"/>

<!-- rosrun ndt_localizer ndt_mapping  -->
  <node pkg="ndt_mapping" type="ndt_localization" name="ndt_localization" output="screen" ns="$(arg namespace)" >
  	<param name="map" value="$(arg map)" />
    <param name="namespace" value="$(arg namespace)" />
    <param name="output_directory" value="$(arg output_directory)" />

  	<param name="resolution" value="$(arg resolution)" />
  	<param name="step_size" value="$(arg step_size)" />
  	<param name="transformation_epsilon" value="$(arg transformation_epsilon)" />
  	<param name="max_iteration" value="$(arg max_iteration)" />
  	<param name="voxel_leaf_size" value="$(arg voxel_leaf_size)" />
  	<param name="min_scan_range" value="$(arg min_scan_range)" />
  	<param name="min_add_scan_shift" value="$(arg min_add_scan_shift)" />
    <param name="min_add_scan_yaw_diff" value="$(arg min_add_scan_yaw_diff)" />
  	
    <param name="initial_x"      value="$(arg initial_x)" />
  	<param name="initial_y"      value="$(arg initial_y)" />
  	<param name="initial_z"      value="$(arg initial_z)" />
  	<param name="initial_roll"   value="$(arg initial_roll)" />
  	<param name="initial_pitch"  value="$(arg initial_pitch)" />
  	<param name="initial_yaw"    value="$(arg initial_yaw)" />

  	<param name="tf_x"      value="$(arg tf_x)" />
  	<param name="tf_y"      value="$(arg tf_y)" />
  	<param name="tf_z"      value="$(arg tf_z)" />
  	<param name="tf_roll"   value="$(arg tf_roll)" />
  	<param name="tf_pitch"  value="$(arg tf_pitch)" />
  	<param name="tf_yaw"    value="$(arg tf_yaw)" />
  </node>

  <!-- <node pkg="tf" type="static_transform_publisher" name="baselink_velodyne_transform" args="0 0 0 0 0 0 /base_link /velodyne 10"/> -->
  
</launch>
